pkg/build/docker_go.go

```diff
@@ -596,8 +626,14 @@ RUN echo "Using go proxy: ${GO_PROXY}" \
 
 {{.DockerfileExtensions.PreSourceCopy}}
 
-# Now copy the rest of the source and run the build.
+# Now copy the rest of the source and run the build. Make sure we backup modfiles first.
+RUN cp ${PLAN_DIR}/go.mod ${PLAN_DIR}/go.sum /tmp/
 COPY . /
+RUN cp /tmp/go.mod /tmp/go.sum ${PLAN_DIR}/
+
+# Copy again the modfiles in case there where overwritten.
+COPY /plan/${MODFILE} ${PLAN_DIR}/go.mod
```

Comment:
Are the lines 629 - 632 necessary? Since lines 635-636 will overwrite the `go.mod` and `go.sum`, I don't see the point in copying the files to `/tmp/`, and then back again.

Reply:
Nice catch, thanks for the note!

It looks like this now:

https://github.com/testground/testground/pull/1338/files#diff-9b5fa505e6278f70a7479cc5cc065c37fd0c374023b8b89b124d948ffc1025e7R630-R632

We do keep a backup of the modfile, instead of overwriting it.
Because earlier in the Dockerfile, when we `go mod download`, the command updated the `go.mod` file with the effective module versions.